packages/block-library/src/cover-image/index.js

```diff
@@ -117,142 +127,168 @@ export const settings = {
-		const toggleParallax = () => setAttributes( { hasParallax: ! hasParallax } );
-		const setDimRatio = ( ratio ) => setAttributes( { dimRatio: ratio } );
+	edit: compose( [
+		withColors( { overlayColor: 'background-color' } ),
+		withNotices,
+	] )(
+		( { attributes, setAttributes, isSelected, className, noticeOperations, noticeUI, overlayColor, setOverlayColor } ) => {
+			const { url, title, align, contentAlign, id, hasParallax, dimRatio } = attributes;
+			const updateAlignment = ( nextAlign ) => setAttributes( { align: nextAlign } );
+			const onSelectImage = ( media ) => {
+				if ( ! media || ! media.url ) {
+					setAttributes( { url: undefined, id: undefined } );
+					return;
+				}
+				setAttributes( { url: media.url, id: media.id } );
+			};
+			const toggleParallax = () => setAttributes( { hasParallax: ! hasParallax } );
+			const setDimRatio = ( ratio ) => setAttributes( { dimRatio: ratio } );
+			const setTitle = ( newTitle ) => setAttributes( { title: newTitle } );
 
-		const style = backgroundImageStyles( url );
-		const classes = classnames(
-			className,
-			contentAlign !== 'center' && `has-${ contentAlign }-content`,
-			dimRatioToClass( dimRatio ),
-			{
-				'has-background-dim': dimRatio !== 0,
-				'has-parallax': hasParallax,
-			}
-		);
+			const style = {
+				...backgroundImageStyles( url ),
+				backgroundColor: overlayColor.color,
+			};
 
-		const controls = (
-			<Fragment>
-				<BlockControls>
-					<BlockAlignmentToolbar
-						value={ align }
-						onChange={ updateAlignment }
-					/>
-					<AlignmentToolbar
-						value={ contentAlign }
-						onChange={ ( nextAlign ) => {
-							setAttributes( { contentAlign: nextAlign } );
-						} }
+			const classes = classnames(
+				className,
+				contentAlign !== 'center' && `has-${ contentAlign }-content`,
+				dimRatioToClass( dimRatio ),
+				{
+					'has-background-dim': dimRatio !== 0,
+					'has-parallax': hasParallax,
+				}
+			);
+
+			const controls = (
+				<Fragment>
+					<BlockControls>
+						<BlockAlignmentToolbar
+							value={ align }
+							onChange={ updateAlignment }
+						/>
+						<AlignmentToolbar
+							value={ contentAlign }
+							onChange={ ( nextAlign ) => {
+								setAttributes( { contentAlign: nextAlign } );
+							} }
+						/>
+						<Toolbar>
+							<MediaUpload
+								onSelect={ onSelectImage }
+								allowedTypes={ ALLOWED_MEDIA_TYPES }
+								value={ id }
+								render={ ( { open } ) => (
+									<IconButton
+										className="components-toolbar__control"
+										label={ __( 'Edit image' ) }
+										icon="edit"
+										onClick={ open }
+									/>
+								) }
+							/>
+						</Toolbar>
+					</BlockControls>
+					{ !! url && (
+						<InspectorControls>
+							<PanelBody title={ __( 'Cover Image Settings' ) }>
+								<ToggleControl
+									label={ __( 'Fixed Background' ) }
+									checked={ hasParallax }
+									onChange={ toggleParallax }
+								/>
+								<PanelColorSettings
```

Comment:
PanelColorSettings includes a small rectangle with the color preview. As a possible improvement, we may apply the opacity CSS to that rectangle so the color preview is more reliable.